### PR TITLE
Accounts in eacl

### PR DESCRIPTION
**Pull request by smallhive**
Refs https://github.com/nspcc-dev/neofs-api/issues/278

The changes require test changes as well.

ACL owner will be changed from
```json
{
    "Owner": {
        "DisplayName": "Nj9FF9jYTsyX2XniTw5Kqy3Zf91nbhdYiM",
        "ID": "033845e8ebc78251029261cbc0e9bae104d2b7cc9fee5c79720a30332d5b3d1aa0"
    },
    "Grants": [
        {
            "Grantee": {
                "DisplayName": "Nj9FF9jYTsyX2XniTw5Kqy3Zf91nbhdYiM",
                "ID": "033845e8ebc78251029261cbc0e9bae104d2b7cc9fee5c79720a30332d5b3d1aa0",
                "Type": "CanonicalUser"
            },
            "Permission": "FULL_CONTROL"
        }
    ]
}
```
to
```json
{
    "Owner": {
        "DisplayName": "Nj9FF9jYTsyX2XniTw5Kqy3Zf91nbhdYiM",
        "ID": "Nj9FF9jYTsyX2XniTw5Kqy3Zf91nbhdYiM"
    },
    "Grants": [
        {
            "Grantee": {
                "DisplayName": "Nj9FF9jYTsyX2XniTw5Kqy3Zf91nbhdYiM",
                "ID": "Nj9FF9jYTsyX2XniTw5Kqy3Zf91nbhdYiM",
                "Type": "CanonicalUser"
            },
            "Permission": "FULL_CONTROL"
        }
    ]
}
```

### CODE REVIEW

File: api/handler/acl.go [hunk 1]
```diff
@@ -493,14 +493,18 @@ func (h *handler) PutBucketPolicyHandler(w http.ResponseWriter, r *http.Request)
 }
 
 func parseACLHeaders(header http.Header, key *keys.PublicKey) (*AccessControlPolicy, error) {
```
Comment (roman-khimov) [inline api/handler/acl.go:495]: Why keeping the key here?

File: api/handler/acl.go [hunk 2]
```diff
@@ -946,16 +954,12 @@ func formRecords(resource *astResource) ([]*eacl.Record, error) {
...
-			targetKeys := make([]ecdsa.PublicKey, 0, len(astOp.Users))
-			for _, user := range astOp.Users {
-				pk, err := keys.NewPublicKeyFromString(user)
-				if err != nil {
-					return nil, errInvalidPublicKey
-				}
-				targetKeys = append(targetKeys, (ecdsa.PublicKey)(*pk))
-			}
+			t := eacl.NewTarget()
 			// Unknown role is used, because it is ignored when keys are set
```
Comment (roman-khimov) [inline api/handler/acl.go:958]: keys?

File: api/handler/acl.go [hunk 3]
```diff
@@ -989,18 +993,38 @@ func addToList(operations []*astOperation, rec eacl.Record, target eacl.Target)
 	if found != nil {
 		if !groupTarget {
 			for _, key := range target.BinaryKeys() {
```
Comment (roman-khimov) [inline api/handler/acl.go:995]: Can be removed, I think. Only S3 gateways sets them and if it only sets accounts then we may not care about keys.

File: api/handler/acl.go [hunk 4]
```diff
@@ -1011,8 +1035,15 @@ func addToList(operations []*astOperation, rec eacl.Record, target eacl.Target)
 		}
 		if !groupTarget {
 			for _, key := range target.BinaryKeys() {
```
Comment (roman-khimov) [inline api/handler/acl.go:1037]: Same here.

File: api/handler/acl.go [hunk 5]
```diff
@@ -1358,13 +1424,15 @@ func permissionToOperations(permission amazonS3Permission) []eacl.Operation {
 }
 
 func encodeObjectACL(log *zap.Logger, bucketACL *layer.BucketACL, bucketName, objectVersion string) *AccessControlPolicy {
+	account := user.NewFromScriptHash(bucketACL.Info.OwnerPublicKey.GetScriptHash()).String()
```
Comment (roman-khimov) [inline api/handler/acl.go:1427]: Why not using the owner account? You no longer need the key.

File: api/handler/acl.go (hunk 3 above)
Comment (smallhive) [reply api/handler/acl.go]: Ok, removed

File: api/handler/acl.go (hunk 4 above)
Comment (smallhive) [reply api/handler/acl.go]: Removed

File: api/handler/acl.go (hunk 5 above)
Comment (smallhive) [reply api/handler/acl.go]: I didn't want to make strong steps with key usage. Replaced with `Owner`

File: api/handler/acl.go (hunk 1 above)
Comment (roman-khimov) [reply api/handler/acl.go:494]: But I think the only thing you do now with this key is getting an account. In which case the key is irrelevant and as S3 never sets any keys into ACLs, why should it care about them during parsing? I do want to make this easier to use in the end, we only have keys in NeoFS itself for compatibility and that can be reconsidered after some transition period.

Comment (smallhive) [reply api/handler/acl.go]: I agree with this point of view. Replaced with `user.ID`

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**